### PR TITLE
Everything operation with type param code review fixes

### PR DESCRIPTION
**Pull request by jdar8**
- [x] Ensure valid resource type in parameter for `_type`
- [x] Add test for invalid resource type in parameter
- [x] Refactor method parameters in IFhirResourceDaoPatient - extract parameter object
- [x] Refactor tests in ResourceProvider<version>Test - automate getting method name
- [x] Remove test duplication - extracted some methods

#3869

### CODE REVIEW

File: hapi-fhir-jpaserver-test-utilities/src/test/java/ca/uhn/fhir/jpa/search/builder/predicate/ResourceLinkPredicateBuilderTest.java [hunk 1]
```diff
@@ -37,22 +37,22 @@ public void init() {
 	@Test
 	public void createEverythingPredicate_withListOfPids_returnsInPredicate() {
 		Condition condition = myResourceLinkPredicateBuilder.createEverythingPredicate("Patient", new ArrayList<>(),
-			1l, 2l); //fixme
+			1l, 2l);
```
Comment (TipzCM) [inline hapi-fhir-jpaserver-test-utilities/src/test/java/ca/uhn/fhir/jpa/search/builder/predicate/ResourceLinkPredicateBuilderTest.java:40]: but the question is, is it fixed? 🤔  Doesn't seem changed

File: hapi-fhir-storage/src/main/java/ca/uhn/fhir/jpa/api/dao/PatientEverythingParameters.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+public final class PatientEverythingParameters {
+	private IPrimitiveType<Integer> theCount;
+	private IPrimitiveType<Integer> theOffset;
+	private DateRangeParam theLastUpdated;
+	private SortSpec theSort;
+	private StringAndListParam theContent;
+	private StringAndListParam theNarrative;
+	private StringAndListParam theFilter;
+	private StringAndListParam theTypes;
```
Comment (TipzCM) [inline hapi-fhir-storage/src/main/java/ca/uhn/fhir/jpa/api/dao/PatientEverythingParameters.java:16]: Would you hate me if i asked for some documentation on these fields?

Maybe it's obvious to you what "theNarrative" is, but i feel like it isn't for me. Same with "theTypes" and "theFilter" etc.

File: hapi-fhir-jpaserver-test-utilities/src/test/java/ca/uhn/fhir/jpa/provider/dstu3/ResourceProviderDstu3Test.java [hunk 3]
```diff
@@ -4776,5 +4791,7 @@ private String toStr(Date theDate) {
 		return new InstantDt(theDate).getValueAsString();
 	}
 
-
+	private String getTestName() {
+		return new Exception().getStackTrace()[1].getMethodName();
```
[the remark this answers is not in the document]
Comment (jdar8) [reply hapi-fhir-jpaserver-test-utilities/src/test/java/ca/uhn/fhir/jpa/provider/dstu3/ResourceProviderDstu3Test.java]: ended up making it a util and replacing all relevant instances

File: hapi-fhir-jpaserver-test-utilities/src/test/java/ca/uhn/fhir/jpa/search/builder/predicate/ResourceLinkPredicateBuilderTest.java (hunk 1 above)
Comment (jdar8) [reply hapi-fhir-jpaserver-test-utilities/src/test/java/ca/uhn/fhir/jpa/search/builder/predicate/ResourceLinkPredicateBuilderTest.java:40]: Ah yes, the fixme was left in there from me when implementing the initial issue. I needed to add a new parameter in to createEverythingPredicate(), and I just quickly added in the new parameter to this test to make the compiler happy. I meant to make sure that adding an empty arraylist as a parameter was ok in this situation (which it is), then I forgot to remove the fixme.

File: hapi-fhir-storage/src/main/java/ca/uhn/fhir/jpa/api/dao/PatientEverythingParameters.java (hunk 2 above)
Comment (jdar8) [reply hapi-fhir-storage/src/main/java/ca/uhn/fhir/jpa/api/dao/PatientEverythingParameters.java]: I added some docs 👍

**Pull request merged**